api/v1alpha1/operator_types.go

```diff
@@ -0,0 +1,122 @@
+	IngressTypeLB IngressType = "LoadBalancer"
+)
+
+// OperatorSpec defines the desired state of ClusterLink Operator.
+type OperatorSpec struct {
+	DataPlane DataPlaneSpec `json:"dataplane,omitempty"`
+	Ingress   IngressSpec   `json:"ingress,omitempty"`
+
+	// +kubebuilder:validation:Enum=trace;info;warning;debug;error;fatal
+	// +kubebuilder:default=info
+	// LogLevel define the ClusterLink components log level.
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// +kubebuilder:default="ghcr.io/clusterlink-net"
+	// ContainerRegistry is the container registry to pull the ClusterLink project images.
+	ContainerRegistry string `json:"containerRegistry,omitempty"`
+	// +kubebuilder:default="latest"
+	// ImageTag represents the version of the ClusterLink project images.
+	ImageTag string `json:"imageTag,omitempty"`
+}
+
+// OperatorStatus defines the observed state of ClusterlLink Operator.
+type OperatorStatus struct {
+	Controlplane ComponentStatus `json:"controlplane,omitempty"`
+	Dataplane    ComponentStatus `json:"dataplane,omitempty"`
+}
+
+// ComponentStatus defines the status of component in ClusterLink.
+type ComponentStatus struct {
+	// +kubebuilder:validation:Enum=none;ready;error;update
+	// +kubebuilder:default=none
+	Status  StatusType `json:"status,omitempty"`
+	Message string     `json:"message,omitempty"`
+}
+
+// DataPlaneSpec defines the desired state of the dataplane components in ClusterLink.
+type DataPlaneSpec struct {
+	// +kubebuilder:validation:Enum=envoy;go
+	// +kubebuilder:default=envoy
+	// Type represents the types of dataplane, supported values "go" or "envoy".
+	Type string `json:"type,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=10
+	// +kubebuilder:default=1
+	// Replicas represents the number of dataplane replicas.
+	Replicas int `json:"replicas,omitempty"`
+}
+
+// IngressSpec defines the type of the ingress component in ClusterLink.
+type IngressSpec struct {
+	// +kubebuilder:validation:Enum=none;LoadBalancer;NodePort
+	// +kubebuilder:default=none
+	// Type represents the type of service to expose ClusterLink deployment, supported values: "LoadBalancer","NodePort", "none".
+	Type IngressType `json:"type,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Operator is the Schema for the ClusterLink operators API.
+type Operator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OperatorSpec   `json:"spec,omitempty"`
+	Status OperatorStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// OperatorList contains a list of Operator.
+type OperatorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Operator `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&Operator{}, &OperatorList{})
+}
```

Comment:
The other projects I've seen use this in `register.go` (i.e. `groupversion_info.go`) instead:
```
var (
	// SchemeBuilder points to a list of functions added to Scheme.
	SchemeBuilder      runtime.SchemeBuilder
	localSchemeBuilder = &SchemeBuilder
	// AddToScheme applies all the stored functions to the scheme.
	AddToScheme = localSchemeBuilder.AddToScheme
)

func init() {
	localSchemeBuilder.Register(addKnownTypes)
}

// Adds the list of known types to api.Scheme.
func addKnownTypes(scheme *runtime.Scheme) error {
	scheme.AddKnownTypes(SchemeGroupVersion,
		&Operator{},
	)
	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
	return nil
}
```